File: ruby/e20.branch.rb
#!/usr/bin/ruby
print "Input a character: " # prompt
STDOUT.flush                # flush unbuffered stream
keypress = STDIN.getc       # acquire single character
 
# test if keypress matches pattern
if keypress =~ /[a-z]/
  puts "Lowercase letter"
elsif keypress =~ /[A-Z]/
  puts "Uppercase letter"
elsif keypress =~ /[0-9]/
  puts "Digit"
else
  puts "Punctuation, whitespace, or other"
end

File: perl/e20.branch.pl
#!/usr/bin/perl -w
print "Input a character: "; $keypress=getc(STDIN);
  
if ($keypress =~ /[[:lower:]]/) {
    print "Lowercase letter\n";
} elsif ($keypress =~ /[[:upper:]]/) {
    print "Uppercase letter\n";
} elsif ($keypress =~ /[[:digit:]]/) {
    print "digit\n";
} else {
    print "Punctuation, whitespace, or other\n";
}
